### PR TITLE
centralization of notifications

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 1]
```diff
@@ -1903,21 +1903,17 @@ public List<ModificationInfos> applyModifications(Network network, UUID networkU
     ** Build variant : sending message to rabbitmq
      */
     public void buildVariant(UUID networkUuid, BuildInfos buildInfos, String receiver) {
-        sendRunBuildMessage(new BuildExecContext(networkUuid, buildInfos, receiver).toMessage(objectMapper));
-    }
-
-    private void sendRunBuildMessage(Message<String> message) {
-        RUN_MESSAGE_LOGGER.debug("Sending message : {}", message);
-        publisher.send("publishBuild-out-0", message);
+        String buildInfosJson;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:1906]: Why do you remove `BuildExecContext::toMessage` ?

File: src/main/java/org/gridsuite/modification/server/service/NotificationService.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    private StreamBridge publisher;
+
+    // Today we don't send notification inside @Transactional block. If this behavior change, we should use @PostCompletion to
+    // make sure that the notification is sent only when all the work inside @Transactional block is done.
+    private void sendMessage(Message<String> message, String bindingName) {
+        OUTPUT_MESSAGE_LOGGER.debug("Sending message : {}", message);
+        publisher.send(bindingName, message);
+    }
+
+    public void emitBuildResult(String payload, String receiver) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NotificationService.java:39]: emitBuildResultMessage ?

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:1906]: BuildExecContext::toMessage build the message that will be sent in notification. That work is done directly in the NotificationService now.

File: src/main/java/org/gridsuite/modification/server/service/NotificationService.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/modification/server/service/NotificationService.java]: ok. done.

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**